### PR TITLE
Do not spam users with welcome message in every commands

### LINKED ISSUE #10870: Remove "Welcome to Cake..." banner for cake console commands.
* [x] enhancement

* CakePHP Version: 3.3.7
* Platform and Target: 7.0.18-0ubuntu0.16.04.1

### What you did
Ran `bin/cake`

### What happened

Output:
```
Welcome to CakePHP v3.3.7 Console
---------------------------------------------------------------
App : src
Path: /var/www/vhosts/xxxxx/htdocs/src/
PHP : 7.0.18-0ubuntu0.16.04.1
---------------------------------------------------------------
```

This banner makes working with the console a bit dirty, as it is output with each command.

### What you expected to happen
Can we save this output for `bin/cake version` output? Or for a special `info` command option?

### PR DESCRIPTION
**Pull request by rchavik**
This is a alternative of #10875

Closes #10870

### CODE REVIEW

**Comment by lorenzo** [comment]
This cannot target master. I use the welcome protected method to show relevant information from my environment.

I would suggest instead that the check for empty args (or equivalent condition) to be done in the _welcome() method directly. So that anyone overriding it still gets the same behavior as before.

**Comment by markstory** [comment]
Another option could be to leave an empty _welcome() method on Shell, and retain the output on the CommandList shell.

**Comment by rchavik** [comment]
Implemented mark's suggestion and retargeted to 3.next.

**Review by markstory — APPROVED** [review]
Looks good to me

**Comment by lorenzo** [comment]
Thanks @rchavik

File: src/Shell/CommandListShell.php [hunk 1]
```diff
@@ -38,6 +38,22 @@ class CommandListShell extends Shell
     public $tasks = ['Command'];
 
     /**
+     * Displays a header for the shell
+     *
+     * @return void
+     */
+    protected function _welcome()
```
Comment (saeideng) [inline src/Shell/CommandListShell.php]: how this works?

Comment (dereuromark) [inline src/Shell/CommandListShell.php]: It overwrites the parent (empty) method.

Comment (saeideng) [inline src/Shell/CommandListShell.php]: thanks

**Pull request merged**